### PR TITLE
Added support for .env.local

### PR DESCRIPTION
**Pull request by ansel1**
By default, will try to load .env.local, then .env.

In order to allow for files which don't exist, Load() only returns the last load error.  This will be a breaking change for code which loads multiple files and relies on Load failing fast.

The alternative would be a new function (TryLoad or something).

#10

### CODE REVIEW

**Comment by joho** [comment]
@ansel1 terribly sorry I haven't reviewed yet, have been pretty busy lately, will try and have a look (and a think about the error handling) over the next day or two.

**Comment by loriopatrick** [comment]
I was about to make PR for the same thing. +1

**Comment by tuvistavie** [comment]
@ansel1's fork is working fine for me, it would be nice if it could be merged.

File: godotenv.go [hunk 1]
```diff
@@ -37,10 +37,8 @@ func Load(filenames ...string) (err error) {
 
 	for _, filename := range filenames {
 		err = loadFile(filename)
-		if err != nil {
-			return // return early on a spazout
-		}
 	}
+	// return the last load error
 	return
```
Comment (tuvistavie) [inline godotenv.go]: The error could be completely ignored if the first load fails and the second succeeds.
It is fine for `.env.local`, but I am not sure it is really a wanted behavior if called with something like

```
godotenv.Load("filenumberone.env", "filenumbertwo.env")
```

Comment (ansel1) [inline godotenv.go]: Yeah, I guess I'm changing the meaning of Load to be "load any of these which exist, in precedence order".  Which is a useful function to have...in my project's tests, I load 3 files like this (.env.local, .env.test, and .env).

I'd suggest either 

1. keeping the new definition, and letting the user call Load() for each file if they care about errors
2. do 1. and create a new "MustLoad" function using the old semantics
3.  revert Load and create a "LoadAny" function with the new semantics.  But if we do this, existing code will break.  They'll call Load() with no args, which will try and load ".env.local" and blow up.  So I'd probably revert filenamesOrDefault() as well, and add a new anyFilenamesOrDefault() function which returns the ".env.local".

Also, I should have done the same thing to Read().

Comment (joho) [inline godotenv.go]: Yeah it's this change in behaviour that I've been laying an egg over. I've been reading the orginal ruby library that inspired this (and whose implementation I try to mirror), and I don't think I noticed before that the magic autoloading is part of `dotenv-rails` vs `dotenv` pure. So it seems like the "right" thing to do is closest to the third option, but it's also the most wrong thing to do.

**Pull request closed without merging**